evalme/video/video.py

```diff
@@ -0,0 +1,133 @@
+from collections import defaultdict
+
+from label_studio_tools.postprocessing.video import extract_key_frames
+
+from evalme.eval_item import EvalItem
+from evalme.image.object_detection import ObjectDetectionEvalItem
+
+
+class VideoEvalItem(EvalItem):
+    SHAPE_KEY = 'videorectangle'
+
+    def iou_over_time(self, pred, per_label=False):
+        """
+        IOU over time for video frames
+
+        :param pred: Predicted VideoEvalItem
+        :param per_label: per label calcu;ation flag
+        :return: float or dict of floats
+        """
+        # prepare results vars for per_label
+        if per_label:
+            results = defaultdict(float)
+            results_count = defaultdict(int)
+        else:
+            results = 0
+            results_count = 0
+        # extract ALL frames from raw data
+        gt_frames_results = self.get_frames()
+        pred_frames_results = pred.get_frames()
+        # check each predicted frame set score with ground truth frame set
+        for pred_frame in pred_frames_results:
```

Comment:
does it compare across all interpolated frames?
we need a function that takes into account ONLY key frames, and compares them to corresponding keyframes or interpolated frames

Reply:
It checks interpolated frames